sematic/resolvers/local_resolver.py

```diff
@@ -108,6 +110,26 @@ def _populate_run_and_artifacts(self, future: AbstractFuture) -> Run:
 
         return run
 
+    def _resolution_will_start(self, future: AbstractFuture):
+        self._populate_run_and_artifacts(future)
+        self._save_graph()
+        if not api_client.resolution_exists(future.id):
+            self._create_resolution(future.id, detached=False)
+        self._update_resolution_status(future.id, ResolutionStatus.RUNNING)
+
+    def _create_resolution(self, root_future_id, detached):
+        api_client.save_resolution(
+            Resolution(
+                root_id=root_future_id,
+                status=ResolutionStatus.SCHEDULED,
+                kind=ResolutionKind.LOCAL,
+                docker_image_uri=None,
+                settings_env_vars={
+                    name: str(value) for name, value in get_all_user_settings().items()
+                },
+            )
+        )
```

Comment:
Can we DRY between `LocalResolver` and `CloudResolver`? The only thing that changes is `kind` is that right?

Maybe store `_kind` as a class property:

```python
class LocalResolver:
    _kind = ResolutionKind.LOCAL
```

Reply:
kind and image. Made it more DRY